js/guide.addingblocks..md

```diff
@@ -0,0 +1,87 @@
+# __How To Add BLocks__
+
+
+This allows developers to add new palettes and blocks to support
+additional functionality after making changes to the
+core code of Turtle Blocks. 
+
+## how to make blocks
+All blocks are present in a simple format.
+
+1) define the type of block
+  
+   var aBlock = new protoBlock('a');
+  
+   here 'a' can be square,trianglemodename,etc.
+
+   This decides what the block will represent
+   
+   
+2)  define the charcateristic of the block
+  
+    aBlock.pallete = palletes.dict['character'];
+    here character can be pitch,tone,rythm,etc.
+
+
+3)  to use general propeties use the protoblock.js file
+    
+    blocks.protoBlockDict['a'] = aBlock;
+
+
+4)  decide wheteher the block is an internal property or not. If you do not have to display the block use-
+   
+     aBlocks.hidden = true;
+     
+     if the block is supposed to be visible , there is no need for this line.
+  
+  
+5)   define the required properties of the block in the format-
+     
+     aBlock.propety();  
+
+  
+  check the available properties on  [protoblock.js](https://github.com/walterbender/turtleblocksjs/blob/master/js/protoblocks.js)
+  and depending on your needs you can add more features and properties.
+
+## Example
+
+    var squareBlock = new ProtoBlock('square');
+    squareBlock.palette = palettes.dict['pitch'];
+    blocks.protoBlockDict['square'] = squareBlock;
+    squareBlock.hidden = true;
+    //.TRANS: square wave
+    squareBlock.staticLabels.push(_('square'));
+    squareBlock.adjustWidthToLabel();
+    squareBlock.oneArgBlock();
+    squareBlock.defaults.push(440);
+  
+## How to add block
+
+ Make your own copy by cloning the official [respository](https://github.com/walterbender/musicblocks.git) 
+ add your block and make a pull request.
+
+
```

Comment:
There are a few subtleties that I don't expect you to get into here: parameter blocks, new block types, new protoblock types, etc. But mention perhaps that these are topics that will be addressed later.